src/message_interface/dtos.rs

```diff
@@ -0,0 +1,200 @@
+    type Error = crate::Error;
+
+    fn try_from(value: &AddressWithAmountDto) -> crate::Result<Self> {
+        Ok(Self {
+            address: value.address.clone(),
+            amount: u64::from_str(&value.amount)
+                .map_err(|_| iota_client::Error::InvalidAmount(value.amount.clone()))?,
+        })
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+/// Dto for address with amount for `send_micro_transaction()`
+pub struct AddressWithMicroAmountDto {
+    /// Bech32 encoded address
+    pub address: String,
+    /// Amount below the minimum storage deposit
+    pub amount: String,
+    /// Bech32 encoded address return address, to which the storage deposit will be returned. Default will use the
+    /// first address of the account
+    pub return_address: Option<String>,
+    /// Expiration in seconds, after which the output will be available for the sender again, if not spent by the
+    /// receiver before. Default is 1 day
+    pub expiration: Option<u32>,
+}
+
+impl TryFrom<&AddressWithMicroAmountDto> for AddressWithMicroAmount {
+    type Error = crate::Error;
+
+    fn try_from(value: &AddressWithMicroAmountDto) -> crate::Result<Self> {
+        Ok(Self {
+            address: value.address.clone(),
+            amount: u64::from_str(&value.amount)
+                .map_err(|_| iota_client::Error::InvalidAmount(value.amount.clone()))?,
+            return_address: value.return_address.clone(),
+            expiration: value.expiration,
+        })
+    }
+}
+
+/// Dto for an account address with output_ids of unspent outputs.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct AddressWithUnspentOutputsDto {
+    /// The address.
+    #[serde(with = "crate::account::types::address_serde")]
+    pub address: AddressWrapper,
+    /// The address key index.
+    #[serde(rename = "keyIndex")]
+    pub key_index: u32,
+    /// Determines if an address is a public or an internal (change) address.
+    pub internal: bool,
+    /// Amount
+    pub amount: String,
+    /// Output ids
+    #[serde(rename = "outputIds")]
+    pub output_ids: Vec<OutputId>,
+}
+
+impl From<&AddressWithUnspentOutputs> for AddressWithUnspentOutputsDto {
+    fn from(value: &AddressWithUnspentOutputs) -> Self {
+        Self {
+            address: value.address.clone(),
+            key_index: value.key_index,
+            internal: value.internal,
+            amount: value.amount.to_string(),
+            output_ids: value.output_ids.clone(),
+        }
+    }
+}
+
+/// Dto for the balance of an account, returned from [`crate::account::handle::AccountHandle::sync()`] and
+/// [`crate::account::handle::AccountHandle::balance()`].
+#[derive(Default, Clone, Debug, Serialize, Deserialize)]
+pub struct AccountBalanceDto {
+    /// Total amount
+    pub total: String,
+    /// Balance that can currently be spend
+    pub available: String,
+    /// Current required storage deposit amount
+    #[serde(rename = "requiredStorageDeposit")]
```

Comment:
You sometimes use `#[serde(rename_all = "camelCase")]` and sometimes do it per field, align!